configure.ac

```diff
@@ -10,12 +14,225 @@ AM_INIT_AUTOMAKE(subdir-objects)
+		    [AC_MSG_RESULT([no])]
+			AC_DEFINE([PVECLIB_DISABLE_VSX], [1], [Disable VSX])
+                )
+            ]
+        )
+    ]
+
+)
+
+##### Compiler default includes Float128 or needs added CFLAGS #####
+
+CFLAGS=$AM_CFLAGS
+AC_MSG_CHECKING([if $CNAME default supports Float128])
+AC_LINK_IFELSE(
+    [AC_LANG_SOURCE([`cat testprograms/test_ppc_F128.c`])],
+    [AC_MSG_RESULT([yes])]
+        AC_SUBST([PVECLIB_FLOAT128_CFLAGS], [""]),
+    [
+	AC_MSG_RESULT([no])
+        CFLAGS+=" $POWER_F128_CFLAGS"
+        AC_MSG_CHECKING([if $CNAME requires -mfloat128 for Float128])
+        AC_LINK_IFELSE(
+            [AC_LANG_SOURCE([`cat testprograms/test_ppc_F128.c`])],
+            [AC_MSG_RESULT([yes])]
+                AC_SUBST([PVECLIB_FLOAT128_CFLAGS], ["$POWER_F128_CFLAGS"]),
+            [
+		AC_MSG_RESULT([no])
+                CFLAGS="$AM_CFLAGS $POWER9_CFLAGS"
+                AC_MSG_CHECKING([if $CNAME requires $POWER9_CFLAGS for Float128])
+                AC_LINK_IFELSE(
+                    [AC_LANG_SOURCE([`cat testprograms/test_ppc_F128.c`])],
+                    [AC_MSG_RESULT([yes])]
+                        AC_SUBST([PVECLIB_FLOAT128PWR9_CFLAGS], ["POWER9_CFLAGS"]),
+                    [AC_MSG_RESULT([no])]
+                        AC_DEFINE([PVECLIB_DISABLE_F128], [1], [Disable Float128])
+                )
+            ]
+        )
+    ]
+)
+
+##### Concatinate FLOAT128 CFLAGS to DEFAULT #####
+
+PVECLIB_DEFAULT_CFLAGS+=" $PVECLIB_FLOAT128_CFLAGS"
+
+AC_MSG_RESULT(["PVECLIB_DEFAULT base target set to '$PVECLIB_DEFAULT_CFLAGS'"])
+
+##### Compiler default|CFLAGS includes Float128 Arithmetic #####
+
+CFLAGS="$AM_CFLAGS $PVECLIB_DEFAULT_CFLAGS"
+AC_MSG_CHECKING([if $CNAME $PVECLIB_DEFAULT_CFLAGS supports Float128 arithmetic])
+AC_LINK_IFELSE(
+	[AC_LANG_SOURCE([`cat testprograms/test_ppc_F128arith.c`])],
+	[AC_MSG_RESULT([yes])]
+		AC_SUBST([PVECLIB_FLOAT128ARITH_CFLAGS], ["$PVECLIB_DEFAULT_CFLAGS"]),
+	[AC_MSG_RESULT([no])]
+		AC_DEFINE([PVECLIB_DISABLE_F128ARITH], [1], [Disable Float128 Arithmetic])
+)
+
+##### Compiler default|CFLAGS includes Float128 math.h #####
+
+CFLAGS="$AM_CFLAGS $PVECLIB_DEFAULT_CFLAGS"
+AC_MSG_CHECKING([if $CNAME $PVECLIB_DEFAULT_CFLAGS supports Float128 math.h])
+AC_LINK_IFELSE(
+	[AC_LANG_SOURCE([`cat testprograms/test_ppc_F128math.c`])],
+	[AC_MSG_RESULT([yes])]
+		AC_SUBST([PVECLIB_FLOAT128MATH_CFLAGS], ["$PVECLIB_DEFAULT_CFLAGS"]),
+	[AC_MSG_RESULT([no])]
+		AC_DEFINE([PVECLIB_DISABLE_F128MATH], [1], [Disable Float128 math.h])
+)
+
+##### Power7 #####
+
+CFLAGS="$POWER7_CFLAGS"
+AC_MSG_CHECKING([if $CNAME supports $CFLAGS (Power7)])
+AC_LINK_IFELSE(
+	[AC_LANG_SOURCE([`cat testprograms/test_ppc_PWR7.c`])],
+	[AC_MSG_RESULT([yes])]
+		AC_SUBST([PVECLIB_POWER7_CFLAGS], [$POWER7_CFLAGS]),
+	[AC_MSG_RESULT([no])]
+		AC_DEFINE([PVECLIB_DISABLE_POWER7], [1], [Disable POWER7])
+)
+
+if test "$retval" != "0"; then
+	AC_SUBST([POWER7_CFLAGS],[""])
+fi
```

Comment:
Yeah, how old of a GCC do you support?

Reply:
I stopped testing AT9 last year. So AT10 (GCC 6.) and later 

Would like to trim it back to the last 3 AT versions.

Technically could go back to AT7.1 (first LE) So GCC 4.9? but the test matrix gets ridiculous.

I can write code or test. Are you volunteering?